### PR TITLE
registry image preflights

### PR DESCRIPTION
**Pull request by divolgin**
Adding Private Image check that will show up when read-only private registry is used.

This uses similar pattern to injected support bundle collectors.  When there is a private registry read-only configured, we add a new collector (`Registry`) and analyzer (`RegistryAnalyze`).   The images are populated from the `knownImages` field in the Installation spec, which includes all images found in release spec and was created for a similar purpose in support bundle.  Image pull secret is also generated from registry settings. 

Screen Shot 2021-04-08 at 5 28 31 PM

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -140,4 +140,7 @@ replace (
 	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.20.2
 	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.20.2
 	k8s.io/sample-controller => k8s.io/sample-controller v0.20.2
+	sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.6.5
 )
+
+replace github.com/replicatedhq/troubleshoot => ../troubleshoot
```
Comment (divolgin) [inline go.mod:146]: to be removed when new troubleshoot release is out

File: pkg/preflight/preflight.go [hunk 2]
```diff
@@ -303,3 +304,47 @@ func CreateRenderedSpec(appID string, sequence int64, origin string, inCluster b
...
+				{
+					Fail: &troubleshootv1beta2.SingleOutcome{
+						When:    "missing > 0",
+						Message: "Application uses images that cannot be found in the private registry",
+					},
+				},
+				{
+					Warn: &troubleshootv1beta2.SingleOutcome{
+						When:    "errors > 0",
+						Message: "Availability of application images in the private registru could not be verified.",
```
Comment (salahalsaleh) [inline pkg/preflight/preflight.go:339]:
```suggestion
						Message: "Availability of application images in the private registry could not be verified.",
```

File: pkg/store/ocistore/app_store.go [hunk 3]
```diff
@@ -211,15 +211,14 @@ func (s *OCIStore) CreateApp(name string, upstreamURI string, licenseData string
...
 	app := apptypes.App{
-		ID:                 id,
-		Name:               name,
-		IconURI:            "",
-		CreatedAt:          time.Now(),
-		Slug:               slugProposal,
-		UpstreamURI:        upstreamURI,
-		License:            licenseData,
-		InstallState:       installState,
-		RegistryIsReadOnly: registryIsReadOnly,
```
Comment (salahalsaleh) [inline pkg/store/ocistore/app_store.go:222]: should be removed from the other store as well

File: pkg/preflight/preflight.go [hunk 4]
```diff
@@ -303,3 +304,47 @@ func CreateRenderedSpec(appID string, sequence int64, origin string, inCluster b
...
+	}
+
+	preflight.Spec.Collectors = append(preflight.Spec.Collectors, &troubleshootv1beta2.Collect{
+		Registry: &troubleshootv1beta2.Registry{
+			Images: images,
+		},
+	})
+	preflight.Spec.Analyzers = append(preflight.Spec.Analyzers, &troubleshootv1beta2.Analyze{
+		Registy: &troubleshootv1beta2.RegistryAnalyze{
+			AnalyzeMeta: troubleshootv1beta2.AnalyzeMeta{
```
Comment (divolgin) [inline pkg/preflight/preflight.go:326]: @AmberAlston need a review for this text in this analyzer.  This is the last row in the screenshot attached in description.

Comment (AmberAlston) [reply pkg/preflight/preflight.go:326]: I'm good with the current description text. I'd like to see "Private Images" updated to "Private Registry Images Available"

**Review by salahalsaleh — APPROVED** [review]

**Review by salahalsaleh — APPROVED** [review]

**Pull request merged**